tasks/kernel_matrix_testing/ci.py

```diff
@@ -6,41 +6,38 @@
 import re
 import tarfile
 import xml.etree.ElementTree as ET
-from typing import TYPE_CHECKING, Any, overload
+from typing import TYPE_CHECKING, overload
+from gitlab.v4.objects import ProjectJob
 
-from tasks.libs.ciproviders.gitlab import Gitlab, get_gitlab_token
+from tasks.libs.ciproviders.gitlab_api import get_gitlab_repo
 
 if TYPE_CHECKING:
     from typing_extensions import Literal
 
     from tasks.kernel_matrix_testing.types import Arch, Component, StackOutput, VMConfig
 
 
-def get_gitlab() -> Gitlab:
-    return Gitlab("DataDog/datadog-agent", str(get_gitlab_token()))
-
-
 class KMTJob:
     """Abstract class representing a Kernel Matrix Testing job, with common properties and methods for all job types"""
 
-    def __init__(self, job_data: dict[str, Any]):
-        self.gitlab = get_gitlab()
-        self.job_data = job_data
+    def __init__(self, job: ProjectJob):
+        self.gitlab = get_gitlab_repo()
+        self.job = job
```

Comment:
`get_gitlab_repo` does an API call each time it's called, so adding it here adds a lot of runtime. We could change the arguments here so that the project instance is passed down and not recreated each time.

```suggestion
    def __init__(self, job: ProjectJob, gitlab: Project):
        self.gitlab = gitlab
        self.job = job
```